vnext/Microsoft.ReactNative/Modules/AccessibilityInfoModule.cpp

```diff
@@ -42,15 +42,28 @@ void AccessibilityInfo::setAccessibilityFocus(double /*reactTag*/) noexcept {
 void AccessibilityInfo::announceForAccessibility(std::string announcement) noexcept {
   m_context.UIDispatcher().Post([context = m_context, announcement = std::move(announcement)] {
     // Windows requires a specific element to announce from. Unfortunately the react-native API does not provide a tag
-    // So we need to add a temporary control to raise the notification event from.
+    // So we need to find something to raise the notification event from.
+    xaml::UIElement element{nullptr};
 
-    auto textBlock = xaml::Controls::TextBlock();
-
-    if (!textBlock) {
-      return;
+    if (react::uwp::IsXamlIsland()) {
+      if (auto xamlroot =
+              winrt::Microsoft::ReactNative::XamlUIService::GetAccessibleRoot(context.Properties().Handle())) {
+        element = xamlroot.Content();
+      }
+      if (!element) {
+        if (auto window = xaml::Window::Current()) {
+          if (element = window.Content()) {
+            element.SetValue(xaml::Automation::AutomationProperties::LandmarkTypeProperty(), winrt::box_value(80002));
```

Comment:
this is not needed. An *app* can do this on the frameworkElement they give us if they so choose, but we shouldn't do it for them

Reply:
Also this code would never execute as it is today, if you're in an island then Window::Current would end up being null in winui 3. It works today against system xaml but that's a bug/behavior that is being fixed in winui3